source/scratch/blocks/operator.cpp

```diff
@@ -153,27 +150,9 @@ Value OperatorBlocks::mathOp(Block &block, Sprite *sprite) {
 }
 
 Value OperatorBlocks::equals(Block &block, Sprite *sprite) {
-    Value value1;
-    Value value2;
-    try {
-        value1 = Scratch::getInputValue(block, "OPERAND1", sprite);
-        value2 = Scratch::getInputValue(block, "OPERAND2", sprite);
-    } catch (...) {
-        return Value(false);
-    }
-
-    if (value1.isNumeric() && value2.isNumeric()) {
-        const double double1 = value1.asDouble();
-        const double double2 = value2.asDouble();
-
-        if (std::isnan(double1) && std::isnan(double2)) return Value(true);
-        return Value(double1 == double2);
-    }
-    std::string string1 = value1.asString();
-    std::string string2 = value2.asString();
-    std::transform(string1.begin(), string1.end(), string1.begin(), ::tolower);
-    std::transform(string2.begin(), string2.end(), string2.begin(), ::tolower);
-    return Value(string1 == string2);
+    Value value1 = Scratch::getInputValue(block, "OPERAND1", sprite);
+    Value value2 = Scratch::getInputValue(block, "OPERAND2", sprite);
+    return Value(value1 == value2);
```

Comment:
You don't need intermediates in this case. I would do:
```suggestion
    return Value(Scratch::getInputValue(block, "OPERAND1", sprite) == Scratch::getInputValue(block, "OPERAND2", sprite));
```